packages/flutter/lib/src/material/material_app.dart

```diff
@@ -88,6 +95,15 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
 
   void didChangeSize(Size size) => setState(() { _size = size; });
 
+  void didChangeLocale(ui.Locale locale) => _updateLocaleState(locale);
```

Comment:
Why not just call _updateLocaleState didChangeLocale, and avoid the indirection?